filters/include/pcl/filters/functor_filter.h

```diff
@@ -76,22 +77,40 @@ class FunctorFilter : public FilterIndices<PointT> {
    */
   void
   applyFilter(Indices& indices) override
+  {
+    applyFilter(executor::best_fit{}, indices);
+  }
+
+  template <typename Executor, typename std::enable_if_t<executor::is_instance_of_base_v<executor::inline_executor, Executor> || executor::is_instance_of_base_v<executor::omp_executor, Executor>, int> = 0>
+  void
+  applyFilter (Executor &&exec, Indices& indices)
   {
     indices.clear();
-    indices.reserve(input_->size());
+    indices.reserve(indices_->size());
+    std::vector<int> keep(indices_->size(), 0);
```

Comment:
Use `vector<bool>` from Boost.Containers

Reply:
That would be thread-safe I assume. Does it have any advantage over using `std::vector<uint8_t>`?

Reply:
and yet here you're using `std::vector<int>`. Not exactly the most space efficient option.